ldap/servers/slapd/modify.c

```diff
@@ -1323,6 +1323,15 @@ op_shared_allow_pw_change(Slapi_PBlock *pb, LDAPMod *mod, char **old_pw, Slapi_M
             rc = -1;
             goto done;
         }
+    } else if (pw_is_pwp_admin(pb, pwpolicy, PWP_ADMIN_OR_ROOTDN)) {
```

Comment:
This is called for direct operation only. Should not we call for both internal/direct ?
Possibly checking that this is
<pre>
slapi_pblock_get(pb, SLAPI_EXT_OP_REQ_OID, &oid);
strcasecmp(oid, EXTOP_PASSWD_OID);
</pre>

Reply:
@tbordaz Sorry I'm not sure what you are wanting to do here.  The code does check both direct/internal ops.  Do you want to make sure the internal op is definitely a "password modify extended op" before checking if it's a password admin?
 
    +    } else if (strcmp(oid, EXTOP_PASSWD_OID) == 0 && pw_is_pwp_admin(pb, pwpolicy, PWP_ADMIN_OR_ROOTDN)) {

Reply:
@tbordaz Ok the OID is not passed back to the server in this case.  I did confirm we need to check if it is a password admin regardless if its an internal operation because a non-root DN user will still trigger the internal operation.  

Anyway I'm not sure I see a need to check the OID, and trying to set it is proving difficult - looks like it getting stripped along the way and I'm not sure it's worth pursuing as I don't see a need for it.